### PR TITLE
added register page components

### LINKED ISSUE #4: Create Register Component
image

- components/register/Register
- Input component will be handled by someone else

### PR DESCRIPTION
**Pull request by fardinzam**
Screenshot 2023-10-14 at 4 50 28 PM

### CODE REVIEW

File: src/app/register/page.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
+import Register from "../../components/register/Register";
+
 const Page = () => {
-  return <div>Page</div>;
+  return <Register name={"Ashley"} />;
```
Comment (shahdivyank) [inline src/app/register/page.js]: dont pass it in

File: src/components/register/Register.jsx [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+          <h1>
+            Hello <span className="text-[#dd8ab8]">{name}</span>, <br /> Create
+            your username!
+          </h1>
+          <Input />
+        </div>
+        <Button
+          text="Get Started"
+          color="lime"
+          className="mr-[5%] h-[90%] my-auto"
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: shouldnt need custom classes for this

File: src/components/register/Register.jsx [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
-  return <div>Register</div>;
+import Button from "@/components/Button";
+import Input from "../Input";
+
+const Register = ({ name }) => {
+  return (
+    <div className="h-screen w-screen text-white text-3xl flex items-center">
+      <div className="w-full flex flex-rows justify-between ml-[5%]">
+        <div className="flex flex-col">
+          <h1>
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: dont use h1 tags here, just do p tags

File: src/components/register/Register.jsx [hunk 4]
```diff
@@ -1,5 +1,25 @@
-const Register = () => {
-  return <div>Register</div>;
+import Button from "@/components/Button";
+import Input from "../Input";
+
+const Register = ({ name }) => {
+  return (
+    <div className="h-screen w-screen text-white text-3xl flex items-center">
+      <div className="w-full flex flex-rows justify-between ml-[5%]">
+        <div className="flex flex-col">
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: flex-col flex is just normal behavior

File: src/components/register/Register.jsx [hunk 5]
```diff
@@ -1,5 +1,25 @@
-const Register = () => {
-  return <div>Register</div>;
+import Button from "@/components/Button";
+import Input from "../Input";
+
+const Register = ({ name }) => {
+  return (
+    <div className="h-screen w-screen text-white text-3xl flex items-center">
+      <div className="w-full flex flex-rows justify-between ml-[5%]">
```
Comment (shahdivyank) [inline src/components/register/Register.jsx]: flex-rows doesnt do anything

File: src/app/register/page.js (hunk 1 above)
Comment (fardinzam) [reply src/app/register/page.js]: I just left an empty const string

File: src/components/Input.jsx [hunk 6]
```diff
@@ -1,5 +1,24 @@
-const Input = () => {
-  return <div>Input</div>;
+const Input = ({ object, setObject, title, label, placeholder, classes }) => {
+  // const handleInput = (e) => {
```
Comment (shahdivyank) [inline src/components/Input.jsx]: remove comments

File: src/components/Input.jsx [hunk 7]
```diff
@@ -1,5 +1,24 @@
...
+  //   setObject({ ...object, [label]: e.target.value });
+  // };
+
+  return (
+    <div className={`flex items-center w-full ${classes}`}>
+      <h6 className="text-xs px-1 z-40 bg-code-black flex fixed ml-3 mt-[-2%]">
+        {title}
+      </h6>
```
Comment (shahdivyank) [inline src/components/Input.jsx]: no h tags

**Comment by shahdivyank** [comment]
let input tag get merged first and then we can merge this

**Comment by fardinzam** [comment]
just pushed input branch

File: src/components/Button.jsx [hunk 8]
```diff
@@ -3,11 +3,11 @@ import { TbArrowRight } from "react-icons/tb";
 import { COLORS } from "@/data/Button";
 import { useState } from "react";
 
-const Next = ({ text, color, onClick }) => {
+const Next = ({ text, color, onClick, className }) => {
```
Comment (shahdivyank) [inline src/components/Button.jsx]: what is the classname for?

File: src/components/register/Register.jsx [hunk 9]
```diff
@@ -1,5 +1,23 @@
...
+    <div className="h-screen w-screen text-white text-3xl flex items-center">
+      <div className="w-full flex flex-row justify-between mx-[5%]">
+        <div>
+          <p>
+            Hello <span className="text-[#dd8ab8]">{name}</span>, <br /> Create
+            your username!
+          </p>
+          <Input label="Username" placeholder="Enter a username" />
+        </div>
+        <div className="h-[90%] my-auto">
```
Comment (shahdivyank) [inline src/components/register/Register.jsx:20]: surely there is a better way to handle this

File: src/components/register/Register.jsx [hunk 10]
```diff
@@ -1,5 +1,23 @@
-const Register = () => {
-  return <div>Register</div>;
+import Button from "@/components/Button";
+import Input from "../Input";
+
+const Register = ({ name }) => {
+  return (
+    <div className="h-screen w-screen text-white text-3xl flex items-center">
+      <div className="w-full flex flex-row justify-between mx-[5%]">
+        <div>
```
Comment (shahdivyank) [inline src/components/register/Register.jsx:9]: this can be a fragment

File: src/app/layout.js [hunk 11]
```diff
@@ -1,21 +1,23 @@
...
 import Navigation from "@/components/Navigation";
+import { usePathname } from "next/navigation";
 
 const inter = Inter({ subsets: ["latin"] });
 
-export const metadata = {
-  title: "Create Next App",
-  description: "Generated by create next app",
-};
+const noNavbarPagePaths = ["/register"];
```
Comment (shahdivyank) [inline src/app/layout.js:11]: do regex like hackathon website

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**